src/components/MyAccount/Settings/AccountSettingsUtils.test.ts

```diff
@@ -1,14 +1,100 @@
+        homeLibraryCode: "sn",
+        id: 2772226,
+        name: "KAHN, VERA RUTH",
+        notificationPreference: "Email",
+        phones: [{ number: "6466600432", type: "t" }],
+      }
+      const patronUpdateBody = {}
+      expect(
+        updatePatronData(originalPatronData, patronUpdateBody)
+      ).toStrictEqual(originalPatronData)
+    })
+    it("can combine patron data and update body with all fields provided", () => {
+      const originalPatronData = {
+        barcode: "23333121538324",
+        emails: ["veggievera@gmail.com"],
+        expirationDate: "2025-03-28",
+        homeLibraryCode: "sn",
+        id: 2772226,
+        name: "KAHN, VERA RUTH",
+        notificationPreference: "Email",
+        phones: [{ number: "6466600432", type: "t" }],
+      }
+      const patronUpdateBody = {
+        emails: ["veraruthkahn@gmail.com", "veggievera@gmail.com"],
+        fixedFields: { 268: { label: "Notice Preference", value: "p" } },
+        homeLibraryCode: "mp   ",
+        phones: [
+          { number: "6466600432", type: "t" },
+          { number: "1234567890", type: "t" },
+        ],
+      }
+      const { id, emails, phones, homeLibraryCode, notificationPreference } =
+        updatePatronData(originalPatronData, patronUpdateBody)
+      expect(emails).toStrictEqual([
+        "veraruthkahn@gmail.com",
+        "veggievera@gmail.com",
+      ])
+      expect(phones).toStrictEqual([
+        { number: "6466600432", type: "t" },
+        { number: "1234567890", type: "t" },
+      ])
+      expect(notificationPreference).toEqual("Phone")
+      expect(homeLibraryCode).toStrictEqual("mp   ")
+      expect(id).toEqual(originalPatronData.id)
+    })
+    it("updates original data when updated data is missing fields", () => {
+      const originalPatronData = {
+        barcode: "23333121538324",
+        emails: ["veggievera@gmail.com"],
+        expirationDate: "2025-03-28",
+        homeLibraryCode: "sn",
+        id: 2772226,
+        name: "KAHN, VERA RUTH",
+        notificationPreference: "Email",
+        phones: [{ number: "6466600432", type: "t" }],
+      }
+      const patronUpdateBody = {
+        fixedFields: { 268: { label: "Notice Preference", value: "p" } },
+        homeLibraryCode: "mp   ",
+      }
+      const { id, emails, phones, homeLibraryCode, notificationPreference } =
+        updatePatronData(originalPatronData, patronUpdateBody)
+      expect(emails).toStrictEqual(originalPatronData.emails)
+      expect(phones).toStrictEqual(originalPatronData.phones)
+      expect(homeLibraryCode).toEqual("mp   ")
+      expect(notificationPreference).toEqual("Phone")
+      expect(id).toEqual(originalPatronData.id)
+    })
+  })
   describe("parsePayload", () => {
-    it.todo("does not submit empty form inputs")
+    it("does not submit empty form inputs", () => {
+      const eventTarget = {
+        emails: { value: "" },
+        phones: { value: "" },
+      }
+      // The payload won't actual
```

Comment:
unfinished sentence?

Reply:
oh gosh I wish I remembered what was supposed to be there... 😅 I'll check the git history for clues.